source/code/cjson/cJSON.c

```diff
@@ -659,7 +659,18 @@ static char *print_object(cJSON *item,int depth,int fmt,printbuffer *p)
 
 /* Get Array size/item / object item. */
 int    cJSON_GetArraySize(cJSON *array)							{cJSON *c=array->child;int i=0;while(c)i++,c=c->next;return i;}
-cJSON *cJSON_GetArrayItem(cJSON *array,int item)				{cJSON *c=array->child;  while (c && item>0) item--,c=c->next; return c;}
+cJSON *cJSON_GetArrayItem(cJSON *array,int item)				
+{
+    cJSON *c = array ? array->child : NULL;
```

Comment:
Not too experienced with C any longer, but I would recommend making this check explicit: `array != NULL`